crates/ruff_python_literal/src/format.rs

```diff
@@ -1020,4 +1039,48 @@ mod tests {
             Err(FormatParseError::InvalidCharacterAfterRightBracket)
         );
     }
+
+    #[test]
+    fn test_format_unicode_escape() {
+        let expected = Ok(FormatString {
+            format_parts: vec![FormatPart::Literal("I am a \\N{snowman}".to_owned())],
+        });
+
+        assert_eq!(FormatString::from_str("I am a \\N{snowman}"), expected);
+    }
```

Comment:
These might be nice as snapshot tests so that we don't have to update them manually in the future:


```suggestion
    fn test_format_unicode_escape() {
        assert_debug_snapshot!(FormatString::from_str("I am a \\N{snowman}"), @"<insta will update this>");
    }
```

in the off chance that the representation of `FormatString` changes.

Reply:
I don't have a strong opinion about using snapshot tests here, but this also makes sense to me 👍 I've updated the snapshots in https://github.com/astral-sh/ruff/pull/21901/commits/99a52eba908762420c0922ff89fcb2e17884a1c9, please let me know what you think 🙏